packages/@adobe/spectrum-css-temp/components/dialog/index.css

```diff
@@ -92,16 +92,16 @@ governing permissions and limitations under the License.
 
 .spectrum-Dialog .spectrum-Dialog-grid {
   display: grid;
-  grid-template-columns: var(--spectrum-dialog-padding-x) auto 1fr auto minmax(0, auto) var(--spectrum-dialog-padding-x);
+  grid-template-columns: var(--spectrum-dialog-padding-x) auto 1fr 1fr 1fr minmax(0, auto) var(--spectrum-dialog-padding-x);
   grid-template-rows: auto var(--spectrum-dialog-padding-y) auto auto 1fr auto var(--spectrum-dialog-padding-y);
   grid-template-areas:
-    "hero hero    hero    hero        hero        hero"
-    ".    .       .       .           .           ."
-    ".    heading header  header      typeIcon    ."
-    ".    divider divider divider     divider     ."
-    ".    content content content     content     ."
-    ".    footer  footer  buttonGroup buttonGroup ."
-    ".    .       .       .           .           .";
+    "hero hero    hero    hero        hero        hero        hero"
+    ".    .       .       .           .           .           ."
+    ".    heading header  header      header      typeIcon    ."
+    ".    divider divider divider     divider     divider     ."
+    ".    content content content     content     content     ."
+    ".    footer  footer  buttonGroup buttonGroup buttonGroup ."
+    ".    .       .       .           .           .           .";
```

Comment:
What's the difference between this and if you'd only changed `auto -> 1fr` instead of adding a new column?

I think we'll need a class that is `spectrum-Dialog--noTypeIcon` that we can use to apply a grid area to the header that lets it take over that space as its own

Reply:
If we switch the `auto -> 1fr` it sort of works. There is extra space for the typeIcon, but the buttons switch to vertical. By adding the extra column the buttons stay horizontal. Keeping this horizontal seems worth an extra column.

Can you explain what you mean with this new class?

Reply:
Yep, a class like this https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/dialog/src/Dialog.tsx#L61
but instead of footer or header, it looks for the typeIcon

if it doesn't exist, then the `.noTypeIcon .spectrum-Dialog-header` selector can be updated to this grid definition
```
grid-area: header / typeIcon
```

or something along those lines, I didn't look it up, going off memory

then we should need to change the 'auto' and we shouldn't need an extra column